File: validate_json.py
import json
import sys

try:
    with open('plant_disease.json', 'r', encoding='utf-8') as f:
        data = json.load(f)
except Exception as e:
    print(f"Error reading file: {e}")
    sys.exit(1)

missing = []
for index, item in enumerate(data):
    if not item.get('name_hi') or not item.get('cause_hi') or not item.get('cure_hi'):
        entry = {
            'index': index,
            'name': item.get('name', 'Unknown'),
            'missing_fields': []
        }
        if not item.get('name_hi'): entry['missing_fields'].append('name_hi')
        if not item.get('cause_hi'): entry['missing_fields'].append('cause_hi')
        if not item.get('cure_hi'): entry['missing_fields'].append('cure_hi')
        missing.append(entry)

if not missing:
    print(f"All {len(data)} entries have Hindi translations.")
else:
    print(f"Missing translations for {len(missing)} entries:")
    print(json.dumps(missing, indent=2, ensure_ascii=False))

File: validate_json.js
const fs = require('fs');
const data = JSON.parse(fs.readFileSync('plant_disease.json', 'utf8'));

let missing = [];
data.forEach((item, index) => {
    if (!item.name_hi || !item.cause_hi || !item.cure_hi) {
        missing.push({
            index,
            name: item.name,
            missing_fields: []
        });
        if (!item.name_hi) missing[missing.length - 1].missing_fields.push('name_hi');
        if (!item.cause_hi) missing[missing.length - 1].missing_fields.push('cause_hi');
        if (!item.cure_hi) missing[missing.length - 1].missing_fields.push('cure_hi');
    }
});

if (missing.length === 0) {
    console.log("All " + data.length + " entries have Hindi translations.");
} else {
    console.log("Missing translations for " + missing.length + " entries:");
    console.log(JSON.stringify(missing, null, 2));
}
